### PR TITLE
Improved namespace completion accuracy

### PR DESCRIPTION
**Pull request by sserbin**
Somewhat fixes changes made in #28 (namely, in c4af64e6e7dc39c9403a47a67f4bb51143020621)

Thanks.

---
This change is [Reviewable](https://reviewable.io/reviews/lvht/phpcd.vim/102)

### CODE REVIEW

File: php/PHPCD.php [hunk 1]
```diff
@@ -989,16 +989,19 @@ public function classmap($pattern)
         $use_fqdn = isset($pattern[0]) && $pattern[0] === "\\";
         $pattern = trim($pattern, "\\");
         $classmap_file = $this->root.'/vendor/composer/autoload_classmap.php';
-        $classmap = get_declared_traits() + get_declared_interfaces()
-            + get_declared_classes();
+        $declaredSymbols = array_merge(get_declared_traits(), get_declared_interfaces(),
```
Comment (lvht) [inline php/PHPCD.php]: what's the difference between array_merge and array sum?

**Comment by lvht** [comment]
just use plugins like deoplete.

File: php/PHPCD.php (hunk 1 above)
Comment (sserbin) [reply php/PHPCD.php]: The difference is vital on numeric keys arrays (like what `get_declared_clasases()` and others returns).
Here's an example:  https://3v4l.org/TVDbX

**Comment by sserbin** [comment]
> just use plugins like deoplete.

I'm not sure what that has to do with proposed changes.

What the pull request does is makes the namespace completion more accurate.

At the moment, when you try to complete the `use` statement, you'd get a very long completion list which would *always* include standard classes (like `RuntimeException`, `DateTime` and others) - the resulting array (in `classmap()`) consists of `get_declared_classes` + matched class names from composer's classmap.
This is because the `$classmap`(which is `get_declared_classes()` + others) are *not* filtered by `matchPattern`

**Comment by lvht** [comment]
fixed at 50e85ece32f9c83f71a32e7a470b78e9f9f90711

Thank you 👍

**Pull request closed without merging**